spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithoutReordering.java

```diff
@@ -202,11 +202,6 @@ public Type primitive(Type.PrimitiveType primitive) {
             "Cannot project decimal with incompatible precision: %s < %s",
             requestedDecimal.precision(), decimal.precision());
         break;
-      case TIMESTAMP:
-        Types.TimestampType timestamp = (Types.TimestampType) primitive;
-        Preconditions.checkArgument(timestamp.shouldAdjustToUTC(),
-            "Cannot project timestamp (without time zone) as timestamptz (with time zone)");
```

Comment:
Was this support unavailable just for ORC? or for all readers?

If it's for all readers, was there no change similar to SparkORCReader or VectorizedSparkORCReader required in those respective readers?

Reply:
Avro and Parquet do not support timestamp natively, both Ts and TsTz are just longs for Avro and Parquet.